# Explotary Data Analysis for New Egg Products

In [22]:
# Load dataset and view basic structure
import pandas as pd
import numpy as np

pd.set_option("display.max_rows", None)
pd.set_option("display.max_seq_items", None)

df=pd.read_csv("filtered_products_newegg.csv")



## Data Inspection and Cleaning

This section examines the structure and quality of the dataset to ensure it is prepared for analysis. The number of rows and columns, data types and missing values are inspected. These checks help identify incomplete, duplicated, or inconsistent records. Rows with missing values are removed, duplicate entries are dropped, and products with zero reviews are filtered out, as they do not contribute meaningful insight. A final structural check confirms that the dataset is complete, consistent, and ready for exploratory analysis

In [23]:
#shape
df.shape

(2605, 10)

In [24]:
# Check structure
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2605 entries, 0 to 2604
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   itemNumber             2605 non-null   object 
 1   class                  2605 non-null   object 
 2   name                   2603 non-null   object 
 3   customerReviewAverage  2605 non-null   float64
 4   customerReviewCount    2605 non-null   int64  
 5   customerTopRated       2605 non-null   bool   
 6   modelNumber            2602 non-null   object 
 7   salePrice              2605 non-null   float64
 8   manufacturer           2605 non-null   object 
 9   isRefurbished          2605 non-null   bool   
dtypes: bool(2), float64(2), int64(1), object(5)
memory usage: 168.0+ KB


In [25]:
# Missing values summary
df.isnull().sum()

itemNumber               0
class                    0
name                     2
customerReviewAverage    0
customerReviewCount      0
customerTopRated         0
modelNumber              3
salePrice                0
manufacturer             0
isRefurbished            0
dtype: int64

In [26]:
# Print the number of rows and columns
print(f"The number of rows= {df.shape[0]}")
print(f"The number of columns= {df.shape[1]}")

The number of rows= 2605
The number of columns= 10


In [27]:
#display the columns
df.columns

Index(['itemNumber', 'class', 'name', 'customerReviewAverage',
       'customerReviewCount', 'customerTopRated', 'modelNumber', 'salePrice',
       'manufacturer', 'isRefurbished'],
      dtype='object')

In [28]:
df.columns.to_list()

['itemNumber',
 'class',
 'name',
 'customerReviewAverage',
 'customerReviewCount',
 'customerTopRated',
 'modelNumber',
 'salePrice',
 'manufacturer',
 'isRefurbished']

In [29]:
# Data Types
df.dtypes

itemNumber                object
class                     object
name                      object
customerReviewAverage    float64
customerReviewCount        int64
customerTopRated            bool
modelNumber               object
salePrice                float64
manufacturer              object
isRefurbished               bool
dtype: object

In [30]:
#remove rows that are empty
df = df.dropna()

In [31]:
# Drop duplicates if any
df.drop_duplicates(inplace=True)

In [32]:
df.isnull().sum()

itemNumber               0
class                    0
name                     0
customerReviewAverage    0
customerReviewCount      0
customerTopRated         0
modelNumber              0
salePrice                0
manufacturer             0
isRefurbished            0
dtype: int64

In [33]:
df = df[df['customerReviewCount'] != 0]

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 618 entries, 0 to 2587
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   itemNumber             618 non-null    object 
 1   class                  618 non-null    object 
 2   name                   618 non-null    object 
 3   customerReviewAverage  618 non-null    float64
 4   customerReviewCount    618 non-null    int64  
 5   customerTopRated       618 non-null    bool   
 6   modelNumber            618 non-null    object 
 7   salePrice              618 non-null    float64
 8   manufacturer           618 non-null    object 
 9   isRefurbished          618 non-null    bool   
dtypes: bool(2), float64(2), int64(1), object(5)
memory usage: 44.7+ KB


## Descriptive Statistics

This section provides an initial exploration of the dataset to understand patterns, distributions, and key characteristics. Numerical features are summarized using measures such as mean, median, quartiles, and standard deviation, while categorical variables are described through frequency counts. These outputs reveal how ratings, review counts, and prices are distributed, and how product categories and manufacturers are represented. The results help identify early trends in customer behavior, product performance, and dataset composition.

In [35]:
# Summary statistics for numeric columns
df.describe()
df.describe(percentiles=[.1, .25, .5, .75, .9])

,customerReviewAverage,customerReviewCount,salePrice
count,618.000000,618.000000,618.000000
mean,4.214725,31.804207,954.188091
std,1.031871,93.610588,883.862430
min,1.000000,1.000000,79.000000
10%,3.000000,1.000000,178.466000
25%,4.000000,1.000000,318.240000
50%,4.500000,3.000000,671.000000
75%,5.000000,20.750000,1379.990000
90%,5.000000,71.000000,1999.990000
max,5.000000,892.000000,7499.000000


In [36]:
# Summary statistics for categorical columns
df.describe(include='object')

,itemNumber,class,name,modelNumber,manufacturer
count,618,618,618,618,618
unique,618,3,615,607,70
top,83-360-831,desktop,HP EliteDesk 800 G9 Mini Desktop Intel Core i7...,FX608JHR-ES73,DELL
freq,1,423,2,4,92


In [37]:
# Univariate Analysis — Numerical
print('Review count Summary:\n ', df['customerReviewCount'].describe())
print('\nReview avg Score Summary:\n ', df['customerReviewAverage'].describe())
print('\n Sale Price Summary:\n ', df['salePrice'].describe())

# Univariate Analysis — Categorical
print('\nManufacturer Summary:\n ', df['manufacturer'].describe())
print('\nclass Summary:\n ', df['class'].describe())
print('\nModel number Summary:\n ', df['modelNumber'].describe())
print('\ncondition Summary:\n ', df['isRefurbished'].describe())

# Value counts help us understand category frequency distribution.
print('\nManufacturer num:\n ', df['manufacturer'].value_counts())
print('\nclass num:\n ', df['class'].value_counts())

Review count Summary:
  count    618.000000
mean      31.804207
std       93.610588
min        1.000000
25%        1.000000
50%        3.000000
75%       20.750000
max      892.000000
Name: customerReviewCount, dtype: float64

Review avg Score Summary:
  count    618.000000
mean       4.214725
std        1.031871
min        1.000000
25%        4.000000
50%        4.500000
75%        5.000000
max        5.000000
Name: customerReviewAverage, dtype: float64

 Sale Price Summary:
  count     618.000000
mean      954.188091
std       883.862430
min        79.000000
25%       318.240000
50%       671.000000
75%      1379.990000
max      7499.000000
Name: salePrice, dtype: float64

Manufacturer Summary:
  count      618
unique      70
top       DELL
freq        92
Name: manufacturer, dtype: object

class Summary:
  count         618
unique          3
top       desktop
freq          423
Name: class, dtype: object

Model number Summary:
  count               618
unique              607
top     

## Detailed Statistical Summaries & Outlier Analysis

This section computes central tendency and variability measures for key numerical variables, including review count, review rating, and sale price. Statistics such as mean, median, mode, variance, and standard deviation describe typical values and the spread of the data. Outlier detection is performed using both the Interquartile Range (IQR) method and z-scores to identify unusually high or low observations. These outliers are important to recognize because they may influence averages or indicate exceptional product cases.

In [38]:

# Numerical summaries
print("Review Count")
print("Mean Review Count:", df['customerReviewCount'].mean())
print("Median Review Count:", df['customerReviewCount'].median())
print("Mode Review Count:", df['customerReviewCount'].mode()[0])
print("Std of Review Count:",df['customerReviewCount'].std() )
print("Variance of Review Count:",df['customerReviewCount'].var() )
print("Min & Max Review Count:",df['customerReviewCount'].min(), df['customerReviewCount'].max() )

# Using IQR (inter quartile range) method
Q1 = df['customerReviewCount'].quantile(.25)
Q3 = df['customerReviewCount'].quantile(.75)
IQR = Q3 - Q1

lower_fence = Q1 - IQR*1.5
upper_fence = Q3 + IQR*1.5
outliers = df[(df['customerReviewCount']<lower_fence)|(df['customerReviewCount']>upper_fence)]
print("Number of outliers Review Count:", outliers.shape[0])

# Using Z-Score method
from scipy import stats
z = stats.zscore(df['customerReviewCount'])
print("Potential outliers (Z > 3):", df[abs(z) > 3].shape[0])

print("\nReview Rating")
print("Mean Review Rating:", df['customerReviewAverage'].mean())
print("Median Review Rating:", df['customerReviewAverage'].median())
print("Mode Review Rating:", df['customerReviewAverage'].mode()[0])
print("Std of Review Rating:",df['customerReviewAverage'].std() )
print("Variance of Review Rating:",df['customerReviewAverage'].var() )
print("Min & Max Review Rating:",df['customerReviewAverage'].min(), df['customerReviewAverage'].max() )

# Using IQR (inter quartile range) method
Q1 = df['customerReviewAverage'].quantile(.25)
Q3 = df['customerReviewAverage'].quantile(.75)
IQR = Q3 - Q1

lower_fence = Q1 - IQR*1.5
upper_fence = Q3 + IQR*1.5
outliers = df[(df['customerReviewAverage']<lower_fence)|(df['customerReviewAverage']>upper_fence)]
print("Number of outliers Review Rating:", outliers.shape[0])

# Using Z-Score method
z = stats.zscore(df['customerReviewAverage'])
print("Potential outliers (Z > 3):", df[abs(z) > 3].shape[0])

print("\nSale Price")
print("Mean Sale Price:", df['salePrice'].mean())
print("Median Sale Price:", df['salePrice'].median())
print("Mode Sale Price:", df['salePrice'].mode()[0])
print("Std of Sale Price:",df['salePrice'].std() )
print("Variance of Sale Price:",df['salePrice'].var() )
print("Min & Max Sale Price:",df['salePrice'].min(), df['salePrice'].max() )

# Using IQR (inter quartile range) method
Q1 = df['salePrice'].quantile(.25)
Q3 = df['salePrice'].quantile(.75)
IQR = Q3 - Q1

lower_fence = Q1 - IQR*1.5
upper_fence = Q3 + IQR*1.5
outliers = df[(df['salePrice']<lower_fence)|(df['salePrice']>upper_fence)]
print("Number of outliers Sale Price:", outliers.shape[0])

# Using Z-Score method
z = stats.zscore(df['salePrice'])
print("Potential outliers (Z > 3):", df[abs(z) > 3].shape[0])



Review Count
Mean Review Count: 31.8042071197411
Median Review Count: 3.0
Mode Review Count: 1
Std of Review Count: 93.6105878310278
Variance of Review Count: 8762.94215407057
Min & Max Review Count: 1 892
Number of outliers Review Count: 97
Potential outliers (Z > 3): 13

Review Rating
Mean Review Rating: 4.214724919093851
Median Review Rating: 4.5
Mode Review Rating: 5.0
Std of Review Rating: 1.0318713651222398
Variance of Review Rating: 1.0647585141592346
Min & Max Review Rating: 1.0 5.0
Number of outliers Review Rating: 50
Potential outliers (Z > 3): 32

Sale Price
Mean Sale Price: 954.1880906148867
Median Sale Price: 671.0
Mode Sale Price: 1999.99
Std of Sale Price: 883.8624300968479
Variance of Sale Price: 781212.7953367054
Min & Max Sale Price: 79.0 7499.0
Number of outliers Sale Price: 18
Potential outliers (Z > 3): 10


## Categorical variable frequency summaries

This section summarizes the frequency of key categorical variables such as manufacturer and product class. Understanding how often each category appears provides insight into dataset balance, representation, and potential sources of bias. These distributions also support later comparisons between groups when analyzing ratings, review activity, and prices.

In [39]:
# Categorical summaries
print('\nManufacturer counts:\n ', df['manufacturer'].value_counts())
print('\nClass Counts:\n ', df['class'].value_counts())


Manufacturer counts:
  manufacturer
DELL                  92
HP                    53
MSI                   45
Apple                 43
SAMSUNG               40
ABS                   31
Lenovo                31
Skytech               30
STORMCRAFT            23
ASUS                  23
AVGPC                 20
iBUYPOWER             18
Hoengager             15
Cobratype             15
MXZ                    9
YEYIAN                 8
IPASON                 7
Thermaltake            6
XIDAX                  5
Acer America           5
Andromeda Insights     5
GIGABYTE               5
Cooler Master          4
CLX                    4
NINGMEI                4
AOACE                  4
minisforum             4
ZOTAC                  4
ACEMAGIC               4
Higole                 4
yeah magic             3
Google                 3
Kamrui                 3
MLOONG                 3
VTG                    3
ZYNEEX                 2
Periphio               2
Mi                     2
HZG          

## Unique Value Counts

This section reports the number of unique values in each column. Unique value counts help determine the variability of different features and identify which fields are useful for grouping or aggregation. For example, columns with many unique values (such as model numbers) behave differently from those with only a few categories (such as class or condition).

In [40]:
#unique value counts and mode detection
print("\nUnique value counts per column:")
print(df.nunique())


Unique value counts per column:
itemNumber               618
class                      3
name                     615
customerReviewAverage     26
customerReviewCount       80
customerTopRated           2
modelNumber              607
salePrice                431
manufacturer              70
isRefurbished              2
dtype: int64


## Exploring Relationships

This section computes grouped summary statistics to compare rating averages, review counts, and sale prices across product classes and manufacturers. Grouped means highlight meaningful differences between product types or brands and reveal trends that help identify which categories tend to perform better in customer ratings, attract more reviews, or fall into different price ranges.

In [41]:
# Group-based summaries

print("Mean Rating by Class:")
print(df.groupby('class')['customerReviewAverage'].mean())

print("\nMean Rating by Manufacturer:")
print(df.groupby('manufacturer')['customerReviewAverage'].mean())

print("\nMean Review Count by Class:")
print(df.groupby('class')['customerReviewCount'].mean())

print("\nMean Review Count by Manufacturer:")
print(df.groupby('manufacturer')['customerReviewCount'].mean())

print("\nMean Sale Price by Class:")
print(df.groupby('class')['salePrice'].mean())

print("\nMean Sale Price by Manufacturer:")
print(df.groupby('manufacturer')['salePrice'].mean())


Mean Rating by Class:
class
desktop    4.389835
laptop     4.122330
phone      3.513043
Name: customerReviewAverage, dtype: float64

Mean Rating by Manufacturer:
manufacturer
8849                  2.000000
ABS                   4.238710
ACEMAGIC              4.825000
ALCPOK                4.900000
AOACE                 4.700000
AOC Monitors          3.700000
AODU                  3.000000
AORUS                 4.000000
ASUS                  4.595652
AVGPC                 4.485000
Acer America          3.980000
Allied Gaming         1.000000
Andromeda Insights    4.840000
Apple                 3.097674
ArsenalPC             5.000000
Auusda                4.600000
BiTECOOL              5.000000
Bigme                 5.000000
Bitspower             5.000000
CLX                   4.150000
Cobratype             3.946667
Cooler Master         4.350000
Corsair               5.000000
CyberpowerPC          4.000000
DELL                  4.197826
GIGABYTE              3.920000
GenMachine         

The grouped statistics reveal that products with higher average ratings generally receive more reviews, supporting the idea that positive experiences encourage customers to leave feedback. Desktops have the highest mean ratings (4.39) and review counts (44), while phones have lower ratings (3.51) and fewer reviews (2.38). Manufacturer-level data shows that top-rated brands such as ACEMAGIC, ALCPOK, and BiTECOOL tend to have higher review counts, though some low-rated manufacturers also appear with few reviews. Overall, the patterns suggest that satisfaction is linked to higher customer engagement through reviews, though sample size differences across classes and manufacturers may influence these results.

## Correlation Analysis

This section calculates a correlation matrix for the numerical variables to explore relationships within the dataset. Correlation values indicate whether variables move together, such as whether higher review counts are associated with higher ratings or whether price is connected to customer engagement. Identifying these relationships completes the exploratory analysis by showing how numerical features interact.

In [42]:
# Correlation matrix

df.corr(numeric_only=True)

,customerReviewAverage,customerReviewCount,customerTopRated,salePrice,isRefurbished
customerReviewAverage,1.000000,0.091807,0.194476,0.094063,-0.181083
customerReviewCount,0.091807,1.000000,0.418584,0.042378,-0.222528
customerTopRated,0.194476,0.418584,1.000000,0.075428,-0.255116
salePrice,0.094063,0.042378,0.075428,1.000000,-0.496126
isRefurbished,-0.181083,-0.222528,-0.255116,-0.496126,1.000000
